R/zchunk_L141.hfc_R_S_T_Y.R

```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
+    EDGAR_HFC32$Non.CO2 <- "HFC32"
+    EDGAR_HFC365mfc$Non.CO2 <- "HFC365mfc"
+    EDGAR_HFC43$Non.CO2 <- "HFC43"
+
+    # Convert year columns to numeric values, then combine into one tibble
+    # define function to convert data to numeric values
+    years_to_numeric <- function(xx){
+      xx[as.character(1970:2000)] <- sapply(xx[as.character(1970:2000)],as.numeric)
+      return(xx)
+    }
+
+    F_gases <- list(EDGAR_HFC125, EDGAR_HFC134a, EDGAR_HFC143a,
+                    EDGAR_HFC152a, EDGAR_HFC227ea, EDGAR_HFC23,
+                    EDGAR_HFC236fa, EDGAR_HFC245fa, EDGAR_HFC32,
+                    EDGAR_HFC365mfc, EDGAR_HFC43)
+
+    # Use function to convert datat to numeric values and combine to one tibble
+    # Map to GCAM countries and technologies
+    F_gases_numeric <- lapply(F_gases, years_to_numeric)
+    L141.EDGAR_HFC <- do.call(bind_rows,F_gases_numeric) %>%
+      left_join_error_no_match(EDGAR_sector %>% select(-IPCC_description), by = "IPCC") %>% # Add Edgar agg_sector
+      mutate(EDGAR_agg_sector = agg_sector) %>% #rename agg_sector to EDGAR_agg_sector
+      mutate(iso = tolower(ISO_A3), ISO_A3 = NULL) %>% # convert to Edgar ISO
+      change_iso_code('rou', 'rom') %>% # Convert Romania iso code to pre-2002 value
+      left_join_error_no_match(iso_GCAM_regID,by = "iso") %>% # Map iso to GCAM region
+      select(GCAM_region_ID, iso, EDGAR_agg_sector,Non.CO2,one_of(as.character(emissions.EDGAR_YEARS)))
+    L141.EDGAR_hfc_R_S_T_Yh.long <- L141.EDGAR_HFC %>%
+      gather(year, emissions, -GCAM_region_ID,-iso,-EDGAR_agg_sector,-Non.CO2) %>%
+      group_by(GCAM_region_ID, EDGAR_agg_sector, Non.CO2,  year) %>%
+      summarize(emissions = sum(emissions))
+
+    # Map to other f-gas sector, which varies by gas
+    L141.EDGAR_hfc_R_S_T_Yh_rest <- L141.EDGAR_hfc_R_S_T_Yh.long %>%
+      filter(EDGAR_agg_sector != "other_f_gases")
+    L141.EDGAR_hfc_R_S_T_Yh_other <- L141.EDGAR_hfc_R_S_T_Yh.long %>%
+      filter(EDGAR_agg_sector == "other_f_gases") %>%
+      ungroup() %>%
+      select(-EDGAR_agg_sector) %>%
+      left_join_error_no_match(other_f_gases, by = c('Non.CO2' = 'Gas')) %>%
+      rename(EDGAR_agg_sector=Sector)
+    L141.EDGAR_hfc_R_S_T_Yh.long <- bind_rows( L141.EDGAR_hfc_R_S_T_Yh_rest,
+                                               L141.EDGAR_hfc_R_S_T_Yh_other ) %>%
+      mutate(year = as.numeric(year))
+
+    # Map Emissions to GCAM technologies
+    L141.hfc_R_S_T_Yh.long <- gcam_fgas_tech %>%
+      repeat_add_columns(tibble::tibble(GCAM_region_ID = unique(GCAM_region_names$GCAM_region_ID))) %>%
+      repeat_add_columns(tibble::tibble(year = emissions.EDGAR_YEARS)) %>%
+      repeat_add_columns(tibble::tibble(Non.CO2 = unique(L141.EDGAR_hfc_R_S_T_Yh.long$Non.CO2))) %>%
+      left_join(L141.EDGAR_hfc_R_S_T_Yh.long,
+                by = c("EDGAR_agg_sector", "GCAM_region_ID", "year", "Non.CO2")) %>% # there should be NAs in the match.Some emissions sectors have zero F gass emissions
```

Comment:
"there should be NAs in the match, as some emissions sectors have zero F gas emissions"